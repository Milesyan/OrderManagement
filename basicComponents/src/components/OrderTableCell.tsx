import {memo} from "react";
import {IOrder, IOrderStatus} from "src/types";
import styles from './OrderTableCell.module.css';

interface IOrderTableCell {
  order: IOrder;
}
const EVENT_STATUS_MAP: {[key in IOrder['event_name']]: IOrderStatus} = Object.freeze({
  'CREATED': 'Created',
  'COOKED': 'Cooked',
  'DRIVER_RECEIVED': 'Driver Received',
  'DELIVERED': 'Delivered',
  'CANCELLED': 'Cancelled'
})
function centsToDollar(cents: number) {
  return (cents/100).toLocaleString("en-US", {style:"currency", currency:"USD"});

}
const OrderTableCell = (props: IOrderTableCell) => {
  const {order} = props;
  return (
    <tr className={styles.cellRoot}>
      <th className={styles.item}>
        {order.item}
      </th>
      <th className={styles.customer}>
        {order.customer}
      </th>
      <th className={styles.price}>
        {centsToDollar(order.price)}
      </th>
      <th className={styles.destination}>
        {order.destination}
      </th>
      <th className={styles.status}>
        {EVENT_STATUS_MAP[order.event_name] ?? order.event_name}
      </th>
    </tr>
  )
}

export default memo(OrderTableCell);

export function OrderTableHeader() {
  return (
    <tr className={`${styles.cellRoot} ${styles.headerCell}`}>
      <th className={styles.item}>
        Item
      </th>
      <th className={styles.customer}>
        Customer Name
      </th>
      <th className={styles.price}>
        Price
      </th>
      <th className={styles.destination}>
        Destination
      </th>
      <th className={styles.status}>
        Order Status
      </th>
    </tr>)
}