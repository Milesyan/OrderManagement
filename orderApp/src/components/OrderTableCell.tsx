import {memo} from "react";
import {IOrder, IOrderStatus} from "src/types";
import styles from './OrderTableCell.module.css';

interface IOrderTableCell {
  order: IOrder;
  setUpdatingData: (order: IOrder) => void
}
const EVENT_STATUS_MAP: {[key in IOrder['event_name']]: IOrderStatus} = Object.freeze({
  'CREATED': 'Created',
  'COOKED': 'Cooked',
  'DRIVER_RECEIVED': 'Driver Received',
  'DELIVERED': 'Delivered',
  'CANCELLED': 'Cancelled'
})
function centsToDollar(cents: number) {
  return (cents/100).toLocaleString("en-US", {style:"currency", currency:"USD"});

}
const OrderTableCell = (props: IOrderTableCell) => {
  const {order} = props;
  const STATUS_COLOR_MAP: {[key in IOrder['event_name']]?: string} = {
    'CREATED': 'transparent',
    'COOKED': 'lightcyan',
    'DRIVER_RECEIVED': 'lightyellow',
    'DELIVERED': 'lightgreen',
    'CANCELLED': 'lightcoral'
  }
  const onClickUpdate = () => {
    props.setUpdatingData(props.order)
  }
  return (
    <tr className={styles.cellRoot}>
      <th className={styles.item}>
        {order.item}
      </th>
      <th className={styles.customer}>
        {order.customer}
      </th>
      <th className={styles.price}>
        {centsToDollar(order.price)}
      </th>
      <th className={styles.destination}>
        {order.destination}
      </th>
      <th className={styles.status} style={{backgroundColor: STATUS_COLOR_MAP[order.event_name]}}>
        {EVENT_STATUS_MAP[order.event_name] ?? order.event_name}
      </th>
      <th className={styles.status}>
        <button onClick={onClickUpdate}>
          Update
        </button>
      </th>
    </tr>
  )
}

export default memo(OrderTableCell, (prev, curr) => prev.order.id === curr.order.id && prev.order.event_name === curr.order.event_name
 && prev.order.customer === curr.order.customer);

export function OrderTableHeader() {
  return (
    <tr className={`${styles.cellRoot} ${styles.headerCell}`}>
      <th className={styles.item}>
        Item
      </th>
      <th className={styles.customer}>
        Customer Name
      </th>
      <th className={styles.price}>
        Price
      </th>
      <th className={styles.destination}>
        Destination
      </th>
      <th className={styles.status}>
        Order Status
      </th>
      <th className={styles.status}>
        Actions
      </th>
    </tr>)
}