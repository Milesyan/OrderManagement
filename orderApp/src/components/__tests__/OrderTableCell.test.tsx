import OrderTableCell, {OrderTableHeader} from '../OrderTableCell';
import TestRenderer from 'react-test-renderer';

test('render OrderTableCell', () => {
  const tableHeader = TestRenderer.create(<OrderTableHeader/>)
  expect(tableHeader).toMatchSnapshot();
  const tableCell = TestRenderer.create(<OrderTableCell
    order={{
      id: 'xxx-order',
      event_name: "DRIVER_RECEIVED",
      price: 9999,
      item: 'example-item',
      customer: 'miles',
      destination: 'destination'
    }}
  />)
  expect(tableCell).toMatchSnapshot();
});