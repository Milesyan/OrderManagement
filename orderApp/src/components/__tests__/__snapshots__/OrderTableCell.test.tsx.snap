// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render OrderTableCell 1`] = `
<tr
  className="cellRoot headerCell"
>
  <th
    className="item"
  >
    Item
  </th>
  <th
    className="customer"
  >
    Customer Name
  </th>
  <th
    className="price"
  >
    Price
  </th>
  <th
    className="destination"
  >
    Destination
  </th>
  <th
    className="status"
  >
    Order Status
  </th>
</tr>
`;

exports[`render OrderTableCell 2`] = `
<tr
  className="cellRoot"
>
  <th
    className="item"
  >
    example-item
  </th>
  <th
    className="customer"
  >
    miles
  </th>
  <th
    className="price"
  >
    $99.99
  </th>
  <th
    className="destination"
  >
    destination
  </th>
  <th
    className="status"
  >
    Driver Received
  </th>
</tr>
`;
